src/node/desktop/src/main/session-launcher.ts

```diff
@@ -133,36 +133,41 @@ export class SessionLauncher {
   sessionStdout: string[] = [];
   sessionStderr: string[] = [];
   private nextSessionUrl?: string;
-  private splash: BrowserWindow;
-  private showSplash = true;
+  private splash: BrowserWindow | undefined;
+  private showSplash = process.env.NODE_ENV !== 'TEST';
 
   constructor(
     private sessionPath: FilePath,
     private confPath: FilePath,
     private filename: FilePath,
     private appLaunch: ApplicationLaunch,
   ) {
-    this.splash = new BrowserWindow({
-      width: 500,
-      height: 500,
-      frame: false,
-      transparent: true,
-      alwaysOnTop: true,
-      center: true,
-      resizable: false,
-      show: false,
-    });
-
-    this.splash.loadURL(SPLASH_WEBPACK_ENTRY).catch((err: unknown) => logger().logError(err));
-    const splashDelay = process.env.RS_SPLASH_DELAY ? parseInt(process.env.RS_SPLASH_DELAY) : 250;
-
-    setTimeoutPromise(splashDelay)
-      .then(() => {
-        if (this.showSplash) {
-          this.splash.show();
-        }
-      })
-      .catch((err) => logger().logError(err));
+    const splashDelay = process.env.RS_SPLASH_DELAY ? parseInt(process.env.RS_SPLASH_DELAY) : 500;
+    if (process.env.RS_NO_SPLASH) {
+      this.showSplash = false;
+    }
+
+    if (splashDelay > 0) {
```

Comment:
It could be checked before setting the timeout but it still needs to be checked when resolving the promise. The `showSplash` check must be done just before showing the splash. It will be set to `false` if the main window opens after the `splashDelay`. This allows the IDE a grace period before showing the splash in case the startup is fast.

Maybe I can combine the `RS_NO_SPLASH` check with the `splashDelay` check and not bother with setting `showSplash=false`.

Reply:
It just made sense to do as you suggested and add the check before the timeout. I've added a comment to make it clear the the `showSplash` check is required before and after the timeout.